### PR TITLE
Fix typos in Competitive programming tutorial

### PR DESCRIPTION
**Pull request by Inego**

### CODE REVIEW

File: pages/docs/tutorials/competitive-programming.md [hunk 1]
```diff
@@ -171,14 +171,14 @@ fun main() {
...
 
-In this dense code, in addition to collection transformations, you can see such handy Kotlin features as local functions,
-and [elvis operator](/docs/reference/null-safety.html#elvis-operator) `?:`
-that allow to express the 
-[idioms](/docs/reference/idioms.html) like "take value if it is positive or else use length" with concise and readable 
-expression like `.takeIf { it >= 0 } ?: s.length`, yet it is perfectly fine with Kotlin to create additional mutable
+In this dense code, in addition to collection transformations, you can see such handy Kotlin features as local functions
+and the [elvis operator](/docs/reference/null-safety.html#elvis-operator) `?:`
+that allow to express 
+[idioms](/docs/reference/idioms.html) like "take the value if it is positive or else use length" with concise and readable
```
Comment (p7nov) [inline pages/docs/tutorials/competitive-programming.md]:
```suggestion
[idioms](/docs/reference/idioms.html) like "take the value if it is positive or else use length" with a concise and readable 
```

File: pages/docs/tutorials/competitive-programming.md [hunk 2]
```diff
@@ -171,14 +171,14 @@ fun main() {
...
-In this dense code, in addition to collection transformations, you can see such handy Kotlin features as local functions,
-and [elvis operator](/docs/reference/null-safety.html#elvis-operator) `?:`
-that allow to express the 
-[idioms](/docs/reference/idioms.html) like "take value if it is positive or else use length" with concise and readable 
-expression like `.takeIf { it >= 0 } ?: s.length`, yet it is perfectly fine with Kotlin to create additional mutable
+In this dense code, in addition to collection transformations, you can see such handy Kotlin features as local functions
+and the [elvis operator](/docs/reference/null-safety.html#elvis-operator) `?:`
+that allow to express 
+[idioms](/docs/reference/idioms.html) like "take the value if it is positive or else use length" with concise and readable 
+expressions like `.takeIf { it >= 0 } ?: s.length`, yet it is perfectly fine with Kotlin to create additional mutable
```
Comment (p7nov) [inline pages/docs/tutorials/competitive-programming.md]:
```suggestion
expression like `.takeIf { it >= 0 } ?: s.length`, yet it is perfectly fine with Kotlin to create additional mutable
```

File: pages/docs/tutorials/competitive-programming.md [hunk 3]
```diff
@@ -206,12 +206,12 @@ specification in the problem statement line by line:
...
 
-Note, that in competitive programming it is customary to give variables shorter names than it is 
-typical in industrial programming practice, since the code is to be written just once and need not be supported thereafter. 
+Note that in competitive programming it is customary to give variables shorter names than it is 
+typical in industrial programming practice, since the code is to be written just once and not supported thereafter. 
 However, these names are usually still mnemonic &mdash; `a` for arrays,
 `i`, `j`, etc for indices, `r`, and `c` for row and column numbers in tables, `x` and `y` for coordinates, etc.
 It is easier to keep the same names for input data as it is given in the problem statement. 
-However, more complex problems require more code to solve and subsequently variable and function names tend to 
+However, more complex problems require more code and, subsequently, variable and function names tend to
```
Comment (p7nov) [inline pages/docs/tutorials/competitive-programming.md]:
```suggestion
However, more complex problems require more code which leads to using longer self-explanatory
```

File: pages/docs/tutorials/competitive-programming.md [hunk 4]
```diff
@@ -206,12 +206,12 @@ specification in the problem statement line by line:
...
-Note, that in competitive programming it is customary to give variables shorter names than it is 
-typical in industrial programming practice, since the code is to be written just once and need not be supported thereafter. 
+Note that in competitive programming it is customary to give variables shorter names than it is 
+typical in industrial programming practice, since the code is to be written just once and not supported thereafter. 
 However, these names are usually still mnemonic &mdash; `a` for arrays,
 `i`, `j`, etc for indices, `r`, and `c` for row and column numbers in tables, `x` and `y` for coordinates, etc.
 It is easier to keep the same names for input data as it is given in the problem statement. 
-However, more complex problems require more code to solve and subsequently variable and function names tend to 
+However, more complex problems require more code and, subsequently, variable and function names tend to 
 become longer and more self-explanatory.
```
Comment (p7nov) [inline pages/docs/tutorials/competitive-programming.md]:
```suggestion
variable and function names.
```

File: pages/docs/tutorials/competitive-programming.md (hunk 2 above)
Comment (Inego) [reply pages/docs/tutorials/competitive-programming.md]: The original version had the noun in the singular form without an article, and you suggest to revert it. Why?

**Comment by Inego** [comment]
@p7nov Thanks! One minor fix you suggested still confuses me a bit though, see my question above.

File: pages/docs/tutorials/competitive-programming.md (hunk 2 above)
Comment (p7nov) [reply pages/docs/tutorials/competitive-programming.md]: Strange thing, on the **Files changed** tab GitHub shows me this part like: `with a concise and readable expressions`. I've checked in your branch and there is no article, so everything is fine.
Thanks!

**Pull request merged**